.expeditor/scripts/finish_release/bump_rustfmt.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This script attempts to find the latest installable nightly rustfmt.
+# When found, it bumps the repo rustfmt version and submits a new PR
+# with the new version pin and any code base formatting changes from the
+# new version.
+
+# Assumptions:
+# 1. This script runs on a x86_64-unknown-linux-gnu target
+# 2. If rustfmt is installable on x86_64-unknown-linux-gnu, the other Tier1
+#    platforms are assumed to also have a working version.
+
+set -Eeuo pipefail
+
+# shellcheck source=.expeditor/scripts/verify/shared.sh
+source .expeditor/scripts/verify/shared.sh
+
+TODAY=$(date '+%Y-%m-%d')
+# pick out the date from a string like: nightly-2019-12-04
+CURRENT_RUSTFMT_DATE=$(awk -F- '{print $2"-"$3"-"$4}' RUSTFMT_VERSION)
+# convert to seconds for easier date comparison later
+CURRENT_RUSTFMT_DATE_S=$(date '+%s' --date "${CURRENT_RUSTFMT_DATE}")
+CHANNEL=nightly
+# a maximum iterator value for searching elligible nightlies
+MAX_ATTEMPTS=90
+
+# if set, this will be the next version
+export FOUND_VERSION
+
+install_rustup
+
+# set minimal profile so only core items + rustfmt are installed
+echo "--- :habicat: Setting minimal profile"
+rustup set profile minimal
+
+for ((i=0; i<MAX_ATTEMPTS; i++)); do
+  # begin searching nightly date versions using current date
+  # as a starting point
+  attempt_date=$(date '+%Y-%m-%d' --date "${TODAY} -${i} day")
+  # use the conversion to seconds for easy date comparison
+  attempt_date_s=$(date '+%s' --date "${attempt_date}")
+  attempt_version="${CHANNEL}-${attempt_date}"
+  echo "--- Trying Nightly Date Version: ${attempt_version}"
+  if rustup toolchain install "${attempt_version}" --component rustfmt; then
+    FOUND_VERSION=${attempt_version}
+    break
+  fi
+  if [ "${attempt_date_s}" -le "${CURRENT_RUSTFMT_DATE_S}" ]; then
+    # Exit early since there is no newer version available than what is
+    # currently pinned in RUSTFMT_VERSION file.
+    echo "--- No newer ${CHANNEL} versions found. Nothing to do."
+    exit 0
+  fi
+done
+
+if [ -z "${FOUND_VERSION}" ]; then
+  # Something is likely wrong if we did not find an installable version.
+  echo "--- :thumbsdown: No candidate versions found within ${MAX_ATTEMPTS}!"
+  exit 1
+fi
+
+# Success on finding a new version
+echo "--- :thumbsup: New installable version of rustfmt found: ${FOUND_VERSION}"
+
+branch="expeditor/bump_rustfmt_${FOUND_VERSION}"
+git checkout -b "${branch}"
+
+# update the version file
+echo "${FOUND_VERSION}" > RUSTFMT_VERSION
+
+# sweep the codebase
+cargo +"$(< RUSTFMT_VERSION)" fmt
+
+# use -u to only add updated files
+git add -u
+git commit -sm "Bump rustfmt to ${FOUND_VERSION}"
+
+# tell expeditor to open a PR
+echo "--- :shipit: Opening a PR"
+open_pull_request
```

Comment:
I believe this helper function is only available when running in Expeditor via a bash action.https://expeditor.chef.io/docs/reference/actions/#open-pull-request

Reply:
Good catch. I've moved the script out of the `finish_release` pipeline and into a [bash action](https://github.com/habitat-sh/habitat/pull/7258/commits/89ab8a0ecf134f05af4a91c586f28abfb18f2374)

Reply:
I think we should keep this in the pipeline.  We're installing rustup and attempting to install rust versions which take a while to execute. The rough guideline I've heard in the past  for `bash action` vs `pipeline step` is will it take more than 10 seconds to run?  which I'm fairly confident that this will violate.  

Looks like I was wrong about open_pull_request anyways :man_facepalming:  https://github.com/habitat-sh/homebrew-habitat/blob/master/.expeditor/scripts/bump_version.sh#L55   open_pull_request is provided by ci-common. Sorry on the confusion here. I'm going to go drink a bucket of coffee.

Reply:
+1 on leaving this in the pipeline, for the same reasons @smacfarlane mentions.

Reply:
Hmm, it does look like that `bump_version.sh` script is called as a [Bash Action](https://github.com/habitat-sh/homebrew-habitat/blob/master/.expeditor/config.yml#L15) though.. 🤦‍♂ Let me think if there's a way to decouple `open_pull_request` and `post_slack_message` from the rest of the code.

Reply:
:man_facepalming:  Sorry @jeremymv2, apparently there is not enough coffee in the world to make me functional.  

I thought I had written something that opens PRs inside a pipeline, I'll see if I can recall what it was or find it.

Reply:
@jeremymv2   I think I was thinking of the script used here: https://github.com/habitat-sh/habitat/blob/master/.expeditor/finish_release.pipeline.yml#L14-L19

Reply:
Alas, I've confirmed with @tduffield that it is only available in a bash action.